url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/notation/ExtendedPositionDescription.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2008-2011, David Garcinuño Enríquez <dagaren@gmail.com>
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package es.dagaren.gladiator.notation;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+
+import es.dagaren.gladiator.representation.BitboardPosition;
+import es.dagaren.gladiator.representation.Colour;
+import es.dagaren.gladiator.representation.GenericPiece;
+import es.dagaren.gladiator.representation.Position;
+import es.dagaren.gladiator.representation.Square;
+
+/**
+ * @author dagaren
+ *
+ */
+public class ExtendedPositionDescription
+{
+   private static Logger logger = Logger.getLogger(ExtendedPositionDescription.class);
+   
+   protected Position position;
+
+   protected Map<String, List<String>> operationsMap = new HashMap<String, List<String>>();
+   
+   
+   public Position getPosition()
+   {
+      return position;
+   }
+
+   public void setPosition(Position position)
+   {
+      this.position = position;
+   }
+   
+   public Map<String, List<String>> getOperationsMap()
+   {
+      return operationsMap;
+   }
+
+   public void setOperationsMap(Map<String, List<String>> operationsMap)
+   {
+      this.operationsMap = operationsMap;
+   }
+
+   public String toString()
+   {
+      String epdString = "";
+      epdString += this.position.toString();
+      epdString += "\n\nOperaciones:\n";
+      
+      Set<String> opcodes = this.operationsMap.keySet();
+      for(String opcode: opcodes)
+      {
+         epdString += "=>" + opcode + ":";
+         
+         List<String> operands = this.operationsMap.get(opcode);
+         for(String operand : operands)
+         {
+            epdString += " " + operand;
+         }
+         epdString += "\n";
+      }
+      
+      return epdString;
+   }
+   
+   public static ExtendedPositionDescription load(String epdString)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.